lib/src/utils.dart

```diff
@@ -626,3 +626,14 @@ Map<K2, V2> mapMap<K1, V1, K2, V2>(
       key(entry.key, entry.value): value(entry.key, entry.value),
   };
 }
+
+/// Makes sure that the given url always ends with a trailing slash
+String checkEndSlash(String url) {
+  if (url.endsWith('/')) return url;
+  return '$url/';
+}
```

Comment:
This is better named something like `ensureTrailingSlash`

Reply:
I've been thinking about the side effects of this...

The commands affected are:
 * `pub publish` (rarely used with 3rd party pub servers)
 * `pub uploader` (rarely used with 3rd party pub servers, and scheduled for deprecation)
 * `pub get` (commonly used by some users with `PUB_HOSTED_URL` or custom pub-server in `pubspec.yaml`).

The impact on `pub publish` and `pub uploader` is pretty breaking, but these have never really worked as intended, so I think that's fine. Well, it's mostly breaking if your custom-pub-server-url included a `/path` part, because that would have been ignored before by `.resolve('/...`

The impact on `pub get` is only that custom-pub-server-urls ending with a slash, will not get double slashes anymore.
I don't think this is too breaking, and we can probably live with that :D